In [1]:
from faker import Faker
import ProfileDict
import ProfileNamedTuple

In [2]:

profile_named_tuple = ProfileNamedTuple.form_profile_named_tuple()

#profile_dict = Dict.form_profile_dict()
profile_dict = ProfileDict.form_profile_dict_from_namedtuple(profile_named_tuple)

In [13]:

# largest blood type
dict_largest_blood_type = ProfileDict.calculate_largest_blood_type(profile_dict)
namedtuple_largest_blood_type = ProfileNamedTuple.calculate_largest_blood_type(profile_named_tuple)
print('largest blood type and time taken by namedtuple is ', namedtuple_largest_blood_type)
print('largest blood type and time taken time taken by dict is ', dict_largest_blood_type)

largest blood type and time taken by namedtuple is  ('A+', 0.0027022361755371094)
largest blood type and time taken time taken by dict is  ('A+', 0.005786895751953125)


In [14]:

# mean current location
dict_mean_current_location = ProfileDict.calculate_mean_current_location(profile_dict)
namedtuple_mean_current_location = ProfileNamedTuple.calculate_mean_current_location(profile_named_tuple)
print('mean current location and time taken by namedtuple is ', namedtuple_mean_current_location)
print('mean current location and time taken time taken by dict is ', dict_mean_current_location)

mean current location and time taken by namedtuple is  ((Decimal('0.21263269825'), Decimal('-1.0908032496')), 0.006278276443481445)
mean current location and time taken time taken by dict is  ((Decimal('0.21263269825'), Decimal('-1.0908032496')), 0.008318901062011719)


In [15]:
# oldest person age
dict_oldest_person_age = ProfileDict.calculate_oldest_person_age(profile_dict)
namedtuple_oldest_person_age = ProfileNamedTuple.calculate_oldest_person_age(profile_named_tuple)
print('oldest person age and time taken by namedtuple is ', namedtuple_oldest_person_age)
print('oldest person age and time taken time taken by dict is ', dict_oldest_person_age)

oldest person age and time taken by namedtuple is  (115.99726214921287, 0.024785995483398438)
oldest person age and time taken time taken by dict is  (115.99726214921287, 0.08483624458312988)


In [17]:

# Average age
dict_avg_age = ProfileDict.calculate_avg_age(profile_dict)
namedtuple_avg_age = ProfileNamedTuple.calculate_avg_age(profile_named_tuple)

print('Average person age and time taken by namedtuple is ', namedtuple_avg_age)
print('Average person age and time taken time taken by dict is ', dict_avg_age)

Average person age and time taken by namedtuple is  (57.625066940451745, 0.017167091369628906)
Average person age and time taken time taken by dict is  (57.625066940451745, 0.022420883178710938)


## Conclusion
### From the logs above the time taken by namedtuple is less than dict
| Task | Time taken by namedtuple | Time taken by dict |
| --- | --- | --- |
| Largest Blood group | 0.0027022361755371094 | 0.005786895751953125 |
| Mean current location | 0.006278276443481445 | 0.008318901062011719 |
| Oldest person age | 0.024785995483398438 | 0.08483624458312988 |
| Average age | 0.017167091369628906 | 0.022420883178710938 |